### ISSUE TITLE
Not compatible with transformer > 4.30

### ISSUE THREAD

**Issue by anitman** [billwuhao/ComfyUI_IndexTTS#32]
考虑修改代码增加对transformer大于4.30的兼容性，目前大部分节点都依赖4.50以上的版本。
```
# SequenceSummary was removed in newer transformers versions, so we implement it here

class SequenceSummary(nn.Module):
    """
    Compute a single vector summary of a sequence hidden states.
    """

    def __init__(self, config):
        super().__init__()
        self.summary_type = getattr(config, "summary_type", "last")
        if self.summary_type == "attn":
            # We should use a standard multi-head attention module with absolute positional embedding for that.
            # Cf. https://github.com/zihangdai/xlnet/blob/master/modeling.py#L253-L276
            # We can use the same initialization with torch.nn.MultiheadAttention
            self.summary = nn.MultiheadAttention(config.n_embd, config.num_attention_heads, 
                                                dropout=config.attn_pdrop, batch_first=True)
        elif self.summary_type == "attn-cls":
            self.summary = nn.Linear(config.n_embd, config.n_embd)
        else:
            # Default case, use a linear layer
            self.summary = nn.Linear(config.n_embd, config.n_embd)

        self.activation = nn.Tanh()
        self.first_dropout = nn.Dropout(getattr(config, "summary_first_dropout", 0.1))
        self.last_dropout = nn.Dropout(getattr(config, "summary_last_dropout", 0.1))

    def forward(self, hidden_states, cls_index=None):
        """
        Compute a single vector summary of a sequence hidden states.

        Args:
            hidden_states (`torch.FloatTensor` of shape `[batch_size, seq_len, hidden_size]`):
                The hidden states of the last layer.
            cls_index (`torch.LongTensor` of shape `[batch_size]` or `[batch_size, ...]` where ... are optional leading dimensions of `hidden_states`, *optional*):
                Used if `summary_type == "cls_index"` and takes the last token of the sequence as classification token.

        Returns:
            `torch.FloatTensor`: The summary of the sequence hidden states.
        """
        if self.summary_type == "last":
            output = hidden_states[:, -1]
        elif self.summary_type == "first":
            output = hidden_states[:, 0]
        elif self.summary_type == "mean":
            output = hidden_states.mean(dim=1)
        elif self.summary_type == "cls_index":
            if cls_index is not None:
                output = hidden_states.gather(1, cls_index.unsqueeze(-1).unsqueeze(-1).expand(-1, -1, hidden_states.size(-1))).squeeze(1)
            else:
                output = hidden_states[:, -1]
        elif self.summary_type == "attn":
            # This is the attention mechanism case (not commonly used)
            output, _ = self.summary(hidden_states, hidden_states, hidden_states)
            output = output.mean(dim=1)
        elif self.summary_type == "attn-cls":
            # Use cls token with attention
            if cls_index is not None:
                cls_token = hidden_states.gather(1, cls_index.unsqueeze(-1).unsqueeze(-1).expand(-1, -1, hidden_states.size(-1))).squeeze(1)
            else:
                cls_token = hidden_states[:, -1]
            output = self.summary(cls_token)
        else:
            # Default to last token
            output = hidden_states[:, -1]

        output = self.first_dropout(output)
        output = self.summary(output) if hasattr(self.summary, 'weight') else output
        output = self.activation(output)
        output = self.last_dropout(output)

        return output

# Original import that was causing issues:
# from transformers.models.gpt2.modeling_gpt2 import SequenceSummary
